src/azure-cli-telemetry/azure/cli/telemetry/components/records_collection.py

```diff
@@ -40,9 +40,7 @@ def snapshot_and_read(self):
         for each in os.listdir(self._cache_dir):
             self._read_file(os.path.join(self._cache_dir, each))
 
-        shutil.rmtree(self._cache_dir,
-                      ignore_errors=True,
-                      onerror=lambda _, p, tr: self._logger.error('Fail to remove file %s', p))
+        shutil.rmtree(self._cache_dir, ignore_errors=True)
```

Comment:
`onerror` is only deprecated in 3.12, but there is no equivalent in earlier versions: https://github.com/python/cpython/issues/112645.

`onerror` is meaningless when `ignore_errors=True`, so I remove it.

Ref: https://docs.python.org/3/library/shutil.html#shutil.rmtree